libnavigation-core/src/androidTest/java/com/mapbox/navigation/core/TripServiceActivityTest.kt

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.navigation.core
+
+import androidx.test.espresso.Espresso.onIdle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import com.mapbox.navigation.core.test.R
+import com.mapbox.navigation.testing.ui.NotificationTestRule
+import com.schibsted.spain.barista.assertion.BaristaEnabledAssertions.assertEnabled
+import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertContains
+import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
+import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class TripServiceActivityTest :
+    NotificationTestRule<TripServiceActivity>(TripServiceActivity::class.java) {
+
+    @Before
+    fun setup() {
+        onIdle()
+    }
+
+    @Test
+    fun checkStartButtonAccessible() {
+        R.id.toggleNotification.let {
+            assertDisplayed(it)
+            assertEnabled(it)
+            assertContains(it, "Start")
+        }
+    }
+
+    @Test
+    fun checkNotificationViewContent() {
```

Comment:
This is not checking the `TripNotification` view content (only `notifyTextView` added in `TripServiceActivity` which IMO is not logic we want to test as it's not part of the SDK). Should we change the test name and add tests to cover the different `TripNotification` use cases? For now we're only "covering" the _Free Drive_ scenario in `checkNotificationContent` https://github.com/mapbox/mapbox-navigation-android/blob/a0481ac141df032300606bb2575d619eef0ac297/libnavigation-core/src/androidTest/java/com/mapbox/navigation/core/TripServiceActivity.kt#L109 BTW that's the key part we need to cover as it's the business logic implemented in `MapboxTripNotification`.

Adding more tests could be ticketed out and we can follow up in a separate PR.

Reply:
removed not related tests